### ISSUE TITLE
Communicate sizing classes throughout forms

### ISSUE THREAD

**Issue by dominikwilkowski** [WestpacCXTeam/GUI-source#223]
Should we include the sizing of the input field? We got four implemented.
- `sm`
- `md` which can be omitted as it's the default view
- `lg`
- `xl`

We need to check all classes implemented and see if they all follow the same naming convention.

**Comment by justinspencer** [comment]
Yes we should. There seems to be a pattern emerging here. We have an element (eg. button, input field, add-on etc). The element can be displayed at different sizes sm, md, lg, xl. They can also be made into block elements (full width) and there is also a responsive option. We should use a consistent approach when communicating these features in the GUI docs.

**Comment by jonnystening** [comment]
We're bringing in a new classing convention going forward... 'xsmall/small/~~medium~~/large/xlarge' to denote appearance sizing and 'xs/sm/md/lg/xl' to denote breakpoint. That way we wont need classes like `.btn-sm-sm` which is uber confusing. This example would be `.btn-small-sm`, meaning "button is sized small at the `sm` breakpoint".

This new naming approach will affect buttons, icons, modals etc.

Icons to begin with (as part of https://github.com/WestpacCXTeam/GUI-source/issues/360), also referenced by Alerts and Buttons (as icons can be placed in either).

**Comment by jonnystening** [comment]
^ 'medium' will be the default and unlikely to be actually used

**Comment by jonnystening** [comment]
The following modules will need updating to utilise this new naming convention...

- [ ] Icons
- [ ] Buttons
- [ ] Input Addons
- [ ] Input
- [ ] Fields
- [ ] Input Groups
- [ ] Modals
- [ ] Switches
- [ ] Badges (buttons)
- [ ] Alerts (icons)
- [ ] Symbols
- [ ] Tooltips (icons)
